tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc

```diff
@@ -2169,6 +2173,52 @@ TEST_F(OpConverterTest, ConvertReshape) {
   }
 }
 
+TEST_P(OpConverterTest1, ConvertShape) {
+  // Get the NodeDef for Transpose.
+  Scope s = Scope::NewRootScope();
+  auto input = ops::Placeholder(s.WithOpName("input"), tf_type);
+  auto shape = ops::Shape(s.WithOpName("my_shape"), input);
+  const NodeDef& node_def = shape.operation.node()->def();
+
+  Status conversion_status =
+      (trt_mode == TrtTestMode::kImplicitBatch)
+          ? errors::InvalidArgument(
+                "Shape is only supported for explicit batch mode.")
+          : Status::OK();
+  std::vector<TestParamBase> test_params = {
+      TestParamBase{{1, 2, 3}, {}, {3}, {}, conversion_status},
+      // Add input as weight (we use non empty param ({1}) to trigger this).
+      TestParamBase{{1, 2, 3}, {}, {3}, {1}, conversion_status},
+  };
+  for (auto p : test_params) {
+    SCOPED_TRACE(p);
+    Reset();
+    int n_elements = 0;
+    if (trt_mode == TrtTestMode::kExplicitBatch && p.param.empty()) {
```

Comment:
Can we do these to improve readability?
(1) add: auto input_is_weight = [](const TestParamBase) {...}
(2) change the if-stmt to
     if (!input_is_weight(p)) { the current else branch code}
(3) move the comment for the current if-then branch to int n_elements = 0. we can also simplify the comment:
 // When the input is a weight, the shape is known at conversion time and the result of the Shape op is a constant layer. As such, the resulting single node network has 0 inputs.